core/src/test/scala/org/broadinstitute/dsde/rawls/entities/EntityServiceSpec.scala

```diff
@@ -262,6 +262,36 @@ class EntityServiceSpec extends AnyFlatSpec with ScalatestRouteTest with Matcher
     }
   }
 
+  it should "fail to rename an attribute name to a name already in use"  in withTestDataServices { services =>
+    val waitDuration = Duration(10, SECONDS)
+    val ex = intercept[RawlsExceptionWithErrorReport] {
+      Await.result(services.entityService.renameAttribute(testData.wsName, testData.pair1.entityType,
+        AttributeName.withDefaultNS("case"), AttributeRename(AttributeName.withDefaultNS("control"))), waitDuration)
+    }
+    ex.errorReport.message shouldBe "AttributeName(default,control) already exists as an attribute name"
+    ex.errorReport.statusCode shouldBe Some(StatusCodes.Conflict)
+  }
 
+  it should "rename an attribute name as long as the selected name is not in use"  in withTestDataServices { services =>
+    val oldAttributeName = AttributeName.withDefaultNS("case")
+    val waitDuration = Duration(10, SECONDS)
+    assertResult(2) {Await.result(services.entityService.renameAttribute(testData.wsName, testData.pair1.entityType,
+      oldAttributeName, AttributeRename(AttributeName.withDefaultNS("newAttributeName"))), waitDuration)}
+
+    // verify there are no longer any attributes under the old attribute name
+    val queryResult = Await.result(services.entityService.listEntities(testData.wsName, testData.pair1.entityType), waitDuration)
+    queryResult map { entity =>
+      assert(!entity.attributes.keySet.contains(oldAttributeName))
```

Comment:
nit: if this test fails, it will result in an error message of "true was not false" or something unhelpful like that, because you're just asserting a boolean value. To get better error messages, try something like:
```scala
      withClue(s"entity ${entity.entityType}/${entity.name} attributes ") {
        entity.attributes.keySet should not contain oldAttributeName
      }
```
using [matchers](https://www.scalatest.org/user_guide/using_matchers) like this helps with test debugging after the fact!